src/app/services/helpers/rk-helper.ts

```diff
@@ -25,9 +25,11 @@ export class RKHelper {
   /**
    * Extrapolate tracks using RungeKutta propagator.
    * @param track Track which is to be extrapolated.
+   * @param inbounds Function which returns true until the passed position is out of bounds, when it returns false. Default just always returns true.
    * @returns An array of track positions.
+
    */
-  public static extrapolateTrackPositions(track: { dparams: any }): any {
+  public static extrapolateTrackPositions(track: { dparams: any },  inbounds: any=function() {return true;}): any {
```

Comment:
The type for `inbounds` here can become - `inbounds: (pos: Vector3) => boolean = (new Vector3()) => true`.